src/libraries/Common/src/System/Net/Http/aspnetcore/Http2/Hpack/DynamicTable.cs

```diff
@@ -89,11 +90,27 @@ public void Resize(int maxSize)
                 _buffer = newBuffer;
                 _removeIndex = 0;
                 _insertIndex = _count;
-                _maxSize = maxSize;
             }
-            else
+
+            var entry = new HeaderField(staticTableIndex, name, value);
+            _buffer[_insertIndex] = entry;
+
+            if (++_insertIndex == _buffer.Length)
+            {
+                _insertIndex = 0;
+            }
+
+            _size += entry.Length;
+            _count++;
+        }
+
+        public void Resize(int maxSize)
```

Comment:
So resize, if growing, doesn't do anything with the actual buffer now? I guess it doesn't matter since we now grow the buffer on demand.
I assume that ASP doesn't have any specific assumptions around the buffer size and this method.

Reply:
Correct. Maybe `Resize` is a bad name for it, it's more `UpdateMaxSize`.
Even before, lowering the maximum wouldn't resize down the underlying buffer, it'd just update the max setting.